### PR TITLE
35 contact picture

### PR DESCRIPTION
**Pull request by ysunkara**

### CODE REVIEW

File: backend/contactsAPI.py [hunk 1]
```diff
@@ -21,7 +26,6 @@ def get_contacts():
     if request.method == 'POST':
         # Create contact
         contact = Contacts(request.get_json()['contact'])
-        contact.save()
```
Comment (AlexMaclean) [inline backend/contactsAPI.py]: Why did you remove this line?

File: backend/contactsAPI.py [hunk 2]
```diff
@@ -39,6 +43,28 @@ def get_contacts():
...
 
 
+@app.route('/img', methods=['POST'])
+def image():
+    if request.method == 'POST':
+        token = request.headers.get('token')
+        _id = request.form.get('_id')
+        user = User().find_by_token(token)
+        image = request.files['file']
+        hashedName = image.split(".", 1)[0] + str(hash(_id)) + str(hash(datetime.now().strftime("%d")))
```
Comment (AlexMaclean) [inline backend/contactsAPI.py]: `image` is not of a type that can be split, when I try to run the code I'm getting a 500. Try `image.filename`

File: backend/contactsAPI.py [hunk 3]
```diff
@@ -39,6 +43,28 @@ def get_contacts():
         return Response(status=204)
 
 
+@app.route('/img', methods=['POST'])
+def image():
+    if request.method == 'POST':
+        token = request.headers.get('token')
+        _id = request.form.get('_id')
+        user = User().find_by_token(token)
```
Comment (AlexMaclean) [inline backend/contactsAPI.py:53]: Doesn't look like `user` is getting used at all, can you confirm the contact id is in the users contact list?

**Review by AlexMaclean — CHANGES_REQUESTED** [review]
Testing this with the front-end I'm getting a bunch of 500s, see my comments about what I think is going wrong.

File: backend/contactsAPI.py (hunk 1 above)
Comment (ysunkara) [reply backend/contactsAPI.py]: Sorry that was an accident when i was redoing the previous implementation to fit the new specs

File: backend/contactsAPI.py [hunk 4]
```diff
@@ -1,9 +1,14 @@
-from flask import Flask, Response, request, jsonify
+from flask import Flask, Response, request, jsonify, url_for, send_from_directory
 from flask_cors import CORS
 
 from database import User, Contacts
+import os
+from datetime import datetime
 
+
+UPLOAD_FOLDER = 'C:/uploads'
```
Comment (baileywickham) [inline backend/contactsAPI.py]: I run linux, and the sever that will host it for the final runs linux, can we make it filesystem independent?

File: backend/contactsAPI.py [hunk 5]
```diff
@@ -39,6 +43,28 @@ def get_contacts():
         return Response(status=204)
 
 
+@app.route('/img', methods=['POST'])
+def image():
+    if request.method == 'POST':
+        token = request.headers.get('token')
+        _id = request.form.get('_id')
```
Comment (baileywickham) [inline backend/contactsAPI.py:50]: What is this line for?

File: backend/contactsAPI.py [hunk 6]
```diff
@@ -39,6 +43,28 @@ def get_contacts():
...
+        user = User().find_by_token(token)
+        image = request.files['file']
+        hashedName = image.split(".", 1)[0] + str(hash(_id)) + str(hash(datetime.now().strftime("%d")))
+        contact = Contacts().find_by_id(_id)
```
Comment (baileywickham) [inline backend/contactsAPI.py:54]: We don't need to find the contact, the database can do it for us. Just use Contact(request.get_json()['contact']) then .save() will update the existing contact.

Comment (baileywickham) [reply backend/contactsAPI.py:54]:
```python
contact = Contacts(request.get_json()['contact'])                 
contact.save()
```
This will either create a new contact, or update the current contact.

Comment (baileywickham) [reply backend/contactsAPI.py:54]: That way you can avoid finding by _id.

File: backend/contactsAPI.py (hunk 4 above)
Comment (baileywickham) [reply backend/contactsAPI.py]: Also, is that in your root directory? We could do something like 
```python
os.path.join(os.getcwd(), 'uploads/')
```
That way python will handle it all behind the scenes.

File: backend/contactsAPI.py (hunk 6 above)
Comment (baileywickham) [reply backend/contactsAPI.py:54]: @ysunkara Nevermind, the API is different than I expected here, you can ignore these comments.

File: backend/contactsAPI.py (hunk 3 above)
Comment (AlexMaclean) [reply backend/contactsAPI.py:53]: This looks better

**Review by AlexMaclean — APPROVED** [review]
After these changes it looks good! Thanks Yogi

**Review by baileywickham — APPROVED** [review]

**Comment by cprghunter** [comment]
LGTM!

**Review by cprghunter — APPROVED** [review]
